app/src/main/java/ch/blinkenlights/android/vanilla/SongTimeline.java

```diff
@@ -725,6 +732,64 @@ else if (delta == SHIFT_PREVIOUS_SONG || delta == SHIFT_NEXT_SONG) {
 		return getSong(0);
 	}
 
+	/**
+	 * Jumps to a previously-created queue
+	 * @author Markil 3
+	 * @since 1.0.86
+	 */
+	public void revertQueue()
+	{
+		ArrayList<Song> timeline = mSongs;
+		synchronized (this)
+		{
+			if (this.songHistory.size() > 0)
+			{
+				/*
+				 * Saves the queue for later
+				 */
+				if (timeline.size() > 0)
+				{
+					this.songHistory.addFirst(new ArrayList<>(timeline));
+				}
+				timeline.clear();
+				timeline.addAll(this.songHistory.removeLast());
```

Comment:
Probably not fatal, but it would be better if we checked for songs which were deleted in the meantime.

See `readState()` in SongTimeline.java